pkg/agent/clean/orphan_bindings.go

```diff
@@ -172,43 +184,29 @@ func (bc *orphanBindingsCleanup) isOrphanBinding(rb rbacv1.RoleBinding) (bool, e
 		return false, nil
 	}
 
-	var prtbNs, prtbName, uid string
-	var foundLabel, hasNewLabel bool
+	var labelFound bool
+	var label, uid string
 	for k, v := range rb.Labels {
 		if v != auth.PrtbInClusterBindingOwner {
 			continue
 		}
-		foundLabel = true
-		if strings.Contains(k, "_") { // only new label will have '_'
-			hasNewLabel = true
-			pieces := strings.Split(k, "_")
-			prtbNs, prtbName = pieces[0], pieces[1]
-		} else {
+		labelFound = true
+		if stringInSet(bc.prtbHashes, k) {
+			label = k
+		} else if stringInSet(bc.prtbUIDs, k) {
 			uid = k
 		}
 	}
 
-	// if there are no matching labels, we don't need to consider this binding
-	if !foundLabel {
-		return false, nil
-	}
-
-	if hasNewLabel {
-		_, err := bc.prtbs.Get(prtbNs, prtbName, metav1.GetOptions{})
-		if err != nil {
-			if k8serrors.IsNotFound(err) {
-				return true, nil
-			}
-		}
-		return false, err
+	if !labelFound {
+		return false, nil // The role binding is not related to any PRTB, so it can't be orphaned.
 	}
 
-	// label is found but is not new, so it must be the legacy label
-	if _, ok := bc.prtbUIDs[uid]; !ok {
-		return true, nil // no prtb with the uid exists, so binding is orphaned
+	if label == "" && uid == "" {
+		logrus.Debugf("Role binding %s doesn't have a corresponding PRTB", rb.Name)
+		return true, nil
 	}
```

Comment:
I don't think that the distinction here between label and UID is super valuable anymore. I would go with something like the following (if you need to keep the last label behavior):

```golang
var isOrphan bool
for k, v := range rb.Labels {
	if v != auth.PrtbInClusterBindingOwner {
		continue
	}
	_, isHashLabel := bc.prtbHashes[k]
	_, isUIDLabel := bc.prtbUIDs[k]
	// if this isn't related to a current label by UID or hash, it is orphaned
	isOrphan = !isHashLabel && !isUIDLabel
}
if isOrphan {
	logrus.Debugf("Role binding %s doesn't have a corresponding PRTB", rb.Name)
}
return isOrphan, nil
```

You can also remove the `isOrphan` calculation and the log statement if you want, but what I put here basically preserves the above and lets you get rid of that new function.